Add kafka SASL plain authentication support (#9584)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

  # CLI flag: -ingest-storage.kafka.write-clients
  [write_clients: <int> | default = 1]

>   # The username to use to authenticate to Kafka using SASL plain mechanism.

Reviewer: ```suggestion
  # The username used to authenticate to Kafka using the SASL plain mechanism.
```

Author: Done in https://github.com/grafana/mimir/pull/9584/commits/1b51166f8d35f5f67bbf8c575b999e03ce0276db

>   # SASL is enabled when both the username and password have been configured.

Reviewer: ```suggestion
  # To enable SASL, configure both the username and password.
```

Reviewer: We try to avoid passive voice in docs.

Author: Done in https://github.com/grafana/mimir/pull/9584/commits/1b51166f8d35f5f67bbf8c575b999e03ce0276db

  # CLI flag: -ingest-storage.kafka.sasl-username
  [sasl_username: <string> | default = ""]

>   # The password to use to authenticate to Kafka using SASL plain mechanism.

Reviewer: ```suggestion
  # The password used to authenticate to Kafka using the SASL plain mechanism.
```

Author: Done in https://github.com/grafana/mimir/pull/9584/commits/1b51166f8d35f5f67bbf8c575b999e03ce0276db

>   # SASL is enabled when both the username and password have been configured.

Reviewer: ```suggestion
  # To enable SASL, configure both the username and password.
```

Author: Done in https://github.com/grafana/mimir/pull/9584/commits/1b51166f8d35f5f67bbf8c575b999e03ce0276db
